### ISSUE TITLE
Cannot detect jump table over writable data

### ISSUE THREAD

**Issue by v1X3Q0** [Vector35/binaryninja-api#6115]
For what its worth, I am seeing a similar problem. For myself, its because at the moment of the `bctr`, the value for the control register cannot be determined. I'm working on the best way I can to broaden the possible values for the submitted control register. In your case, it seems like the r4 register's value may not be able to be determined, while in mine its r0. Without it being able to get a solid for possible reg values, I'm guessing its just gonna assume that the jump table has infinite scope and eat it.

Temporarily resolution: https://github.com/v1X3Q0/jump-table-branch-editor

<img width="410" alt="Captura de pantalla 2024-11-11 a la(s) 5 13 20 p m" src="https://github.com/user-attachments/assets/86c3ea02-1697-4dda-805f-2ae5957cf422">

For what you can see that in this case, r0 is supposed to be something from index void**[0-7], but binja is considering that the value is `undetermined` for it.

<img width="362" alt="Captura de pantalla 2024-11-11 a la(s) 5 17 45 p m" src="https://github.com/user-attachments/assets/194a1975-0643-464b-8ffc-6b8a49bfc458">

_Originally posted by @v1X3Q0 in https://github.com/Vector35/binaryninja-api/discussions/2270#discussioncomment-11219294_

**Comment by xusheng6** [comment]
Is it possible to share the binary with us?

**Comment by v1X3Q0** [comment]
Absolutely! I've been using a gamecube binary that I shared with @galenbwill 

[Clonewars.zip](https://github.com/user-attachments/files/17718576/Clonewars.zip)

Feel free to hop on over to `0x801a6bc0` as shown in the images. You will see other blocks where disassembly failed in the middle of the binary for similar reasons.

for what its worth, you MAY have to test with https://github.com/v1X3Q0/binaryninja-api/ using the architecture ppc_ps when you open the image. If not, a ton of the disassembly will fail. You'll just have to build and copy in the ppc plugin.

**Comment by xusheng6** [comment]
The dataflow analysis is correct up until the selected line:

![Screenshot 2024-11-19 at 11 54 44 AM](https://github.com/user-attachments/assets/27cdc99c-0ac9-4037-b039-5520e94d0bf4)

r3 has the ConstantPointerValue 0x8035b000, and r0 has UnsignedRangeValue {0x0 : 0x1c: 0x4}. However, we are not calculating the value of `[r3 + r0].d`. If we look at the target area, it looks valid pointers: 

![Screenshot 2024-11-19 at 11 57 21 AM](https://github.com/user-attachments/assets/d13c1a40-cb6d-4da8-bff1-77a8821a4f7a)

The reason for the failure is that the target area is writable, which makes the analysis not creating the jump table

Here is a workaround for now:

1. Navigate to `0x8035b000`, and change the type of the data variable to `void const* const `@3175`[0x8]`
2. (Optional) undefine the functions at 0x801b0f9c, etc (which are targets of the jump table), so that the code shows up in the switch-case, rather than being a tailcall

**Comment by xusheng6** [comment]
It looks correct after applying my changes 

![Screenshot 2024-11-19 at 12 04 50 PM](https://github.com/user-attachments/assets/175b18a0-5d2c-44c2-be87-66d8c6202403)

**Comment by v1X3Q0** [comment]
That definitely did it! Thank you, is that something that will be merged into dev soon? Nice solution!

**Comment by xusheng6** [comment]
> That definitely did it! Thank you, is that something that will be merged into dev soon? Nice solution!

Sorry I am not sure what you are referring to when you say "be merged into dev" soon. I did not propose a possible way for it yet. Also, the chances are we will leave it as is since if we relax it, it will cause other issues

**Comment by v1X3Q0** [comment]
I see I see... ah well, I appreciate the help than. That's a neat solution. I'll see if someday I can have a solution, but in the meantime thank you.

**Comment by v1X3Q0** [comment]
Maybe someday it will present itself as more annoying, til then thank you.

**Comment by xusheng6** [comment]
I am re-opening this for internal disucssion
